### PR TITLE
feat: use a Github App instead of a github bot user for permissions-report

### PR DESCRIPTION
**Pull request by lemeurherve**
Part of https://github.com/jenkins-infra/helpdesk/issues/2788

A Github App `jenkins-infra-reports` has been created and installed on https://github.com/jenkinsci/ with one permission, "Metadata" as "Read-only", on all repositories.

I've added a parameter to the GraphQL query to return only public repositories.

<strike>@daniel-beck do you know if only public repositories should be queried? It's used for a public view of the plugins so it seems to be the case for me but I don't know if it's used elsewhere.</strike>

### CODE REVIEW

**Comment by timja** [comment]
> do you know if only public repositories should be queried? It's used for a public view of the plugins so it seems to be the case for me but I don't know if it's used elsewhere.

there's only public repoes in @jenkinsci

**Review by timja — APPROVED** [review]
I would probably use these var names but non blocking:

- `GITHUB_APP_ID` - 
- `GITHUB_APP_PRIVATE_KEY`

this is how github ui displays it:
image

File: Jenkinsfile [hunk 1]
```diff
@@ -103,14 +103,16 @@ pipeline {
 						label 'docker&&linux'
 					}
 					environment {
-						GITHUB_API = credentials('github-token')
+						GITHUB_PRIVATE_KEY_B64 = credentials('githubapp-jenkins-infra-reports-private-key-b64')
+						GITHUB_APP_IDENTIFIER = credentials('githubapp-jenkins-infra-reports-app-identifier')
+						GITHUB_ORG_NAME = "jenkinsci"
```
Comment (daniel-beck) [inline Jenkinsfile:108]: While this feels like the correct thing to do, I'd not do this to not accidentally copy it for another org for which it is not true that there are only public repos.

Comment (lemeurherve) [reply Jenkinsfile:108]: I've added `privacy: PUBLIC` l96 to avoid this potential mistake.

Comment (timja) [reply Jenkinsfile:108]: I think Daniels point was to not do that, there’s no added benefit and only risk if it gets copied elsewhere

Comment (lemeurherve) [reply Jenkinsfile:108]: It served me for my tests but you are right, no point in keeping it now.

Comment (daniel-beck) [reply Jenkinsfile:108]: `privacy: PUBLIC` looks like a good solution 👍

**Pull request merged**